ee/se/modules/380-metallb/hooks/discovery_l2_load_balancers.go

```diff
@@ -0,0 +1,265 @@
+	}, nil
+}
+
+func applyServiceFilter(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	var service v1.Service
+
+	err := sdk.FromUnstructured(obj, &service)
+	if err != nil {
+		return nil, err
+	}
+
+	if service.Spec.Type != v1.ServiceTypeLoadBalancer {
+		// we only need service of LoadBalancer type
+		return nil, nil
+	}
+
+	var externalIPsCount = 1
+	if externalIPsCountStr, ok := service.Annotations[keyAnnotationExternalIPsCount]; ok {
+		if externalIP, err := strconv.Atoi(externalIPsCountStr); err == nil {
+			if externalIP > 1 {
+				externalIPsCount = externalIP
+			}
+		}
+	}
+
+	var desiredIPs []string
+	if DesiredIPsStr, ok := service.Annotations[l2LoadBalancerIPsAnnotate]; ok {
+		desiredIPs = strings.Split(DesiredIPsStr, ",")
+	}
+
+	var lbAllowSharedIP string
+	if lbAllowSharedIPStr, ok := service.Annotations[lbAllowSharedIPAnnotate]; ok {
+		lbAllowSharedIP = lbAllowSharedIPStr
+	}
+
+	var loadBalancerClass string
+	if service.Spec.LoadBalancerClass != nil {
+		loadBalancerClass = *service.Spec.LoadBalancerClass
+	}
+
+	var assignedLoadBalancerClass string
+	for _, condition := range service.Status.Conditions {
+		if condition.Type == "network.deckhouse.io/LoadBalancerClass" {
+			assignedLoadBalancerClass = condition.Message
+			continue
+		}
+	}
+
+	internalTrafficPolicy := v1.ServiceInternalTrafficPolicyCluster
+	if service.Spec.InternalTrafficPolicy != nil {
+		internalTrafficPolicy = *service.Spec.InternalTrafficPolicy
+	}
+
+	return ServiceInfo{
+		Name:                      service.GetName(),
+		Namespace:                 service.GetNamespace(),
+		LoadBalancerClass:         loadBalancerClass,
+		AssignedLoadBalancerClass: assignedLoadBalancerClass,
+		ExternalIPsCount:          externalIPsCount,
+		Ports:                     service.Spec.Ports,
+		ExternalTrafficPolicy:     service.Spec.ExternalTrafficPolicy,
+		InternalTrafficPolicy:     internalTrafficPolicy,
+		Selector:                  service.Spec.Selector,
+		ClusterIP:                 service.Spec.ClusterIP,
+		PublishNotReadyAddresses:  service.Spec.PublishNotReadyAddresses,
+		DesiredIPs:                desiredIPs,
+		LbAllowSharedIP:           lbAllowSharedIP,
+	}, nil
+}
+
+func applyMetalLoadBalancerClassFilter(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	var metalloadbalancerclass MetalLoadBalancerClass
+
+	err := sdk.FromUnstructured(obj, &metalloadbalancerclass)
+	if err != nil {
+		return nil, err
+	}
+
+	interfaces := []string{}
+	if len(metalloadbalancerclass.Spec.L2.Interfaces) > 0 {
+		interfaces = metalloadbalancerclass.Spec.L2.Interfaces
+	}
+
+	return MetalLoadBalancerClassInfo{
+		Name:         metalloadbalancerclass.Name,
+		AddressPool:  metalloadbalancerclass.Spec.AddressPool,
+		Interfaces:   interfaces,
+		NodeSelector: metalloadbalancerclass.Spec.NodeSelector,
+		IsDefault:    metalloadbalancerclass.Spec.IsDefault,
+	}, nil
+}
+
+func handleLoadBalancers(input *go_hook.HookInput) error {
```

Comment:
```suggestion
func handleL2LoadBalancers(input *go_hook.HookInput) error {
```
In future there will also be handleBGPLoadBalancers.